ui/static/ui/js/learning_resources.jsx

```diff
@@ -1,34 +1,63 @@
 define('learning_resources', [
-  'reactaddons', 'jquery', 'lodash', 'utils'], function (
+  'reactaddons', 'jquery', 'lodash', 'utils', 'select2'], function (
   React, $, _, Utils) {
   'use strict';
 
   var StatusBox = Utils.StatusBox;
 
   var VocabularyOption = React.createClass({
-    render: function () {
+    applySelect2: function () {
+      //this function can be used only if the component has been mounted
       var options = _.map(this.props.terms, function (term) {
-        return <option key={term.slug}
-                       value={term.slug}>
-          {term.label}
-        </option>;
+        return {
+          id: term.slug,
+          text: term.label
+        };
       });
-
+      var thiz = this;
+      var isMultiTerms = this.props.vocabulary.multi_terms;
+      //not allowing the clear for the multi terms dropdown
+      var allowClear = !isMultiTerms;
+      //apply select2 to the right elements
+      $(React.findDOMNode(this)).find('select').select2({
+        multiple: isMultiTerms,
+        data: options,
+        placeholder: "Select a value for " + this.props.vocabulary.name,
+        allowClear: allowClear,
+        theme: "bootstrap",
+      })
+      .val(this.props.selectedTerms)
+      .trigger('change')
+      .on('change', function () {
+        var selectValues = [].concat($(this).val());
```

Comment:
it's used because the `$(this).val()` on a select with single value does not return a list, so I want a generic way to transform the result to a list